circuit/src/verifier/transcript/mod.rs

```diff
@@ -173,24 +174,28 @@ where
 
 	/// Read an elliptic curve point.
 	fn read_ec_point(&mut self) -> Result<Halo2LEcPoint<C, L, P>, snark_verifier::Error> {
-		let mut compressed = C::Repr::default();
-		self.reader.read_exact(compressed.as_mut()).map_err(|err| {
-			VerifierError::Transcript(
-				err.kind(),
-				"invalid field element encoding in proof".to_string(),
-			)
-		})?;
+		let mut x: Option<Integer<_, _, NUM_LIMBS, NUM_BITS, P>> = None;
+		let mut y: Option<Integer<_, _, NUM_LIMBS, NUM_BITS, P>> = None;
 
-		let point: C = Option::from(C::from_bytes(&compressed)).ok_or_else(|| {
-			VerifierError::Transcript(
-				ErrorKind::Other,
-				"invalid point encoding in proof".to_string(),
-			)
-		})?;
-		let coordinates = point.coordinates().unwrap();
+		let _ = self.reader.as_mut().and_then(|reader| {
+			let mut compressed = C::Repr::default();
+			if reader.read_exact(compressed.as_mut()).is_err() {
+				return Value::unknown();
+			}
+			let coords = C::from_bytes(&compressed).unwrap().coordinates().unwrap();
+			x = Some(Integer::<_, _, NUM_LIMBS, NUM_BITS, P>::from_w(*coords.x()));
+			y = Some(Integer::<_, _, NUM_LIMBS, NUM_BITS, P>::from_w(*coords.y()));
+			let value = Option::<C>::from(C::from_bytes(&compressed))
+				.ok_or_else(|| {
+					VerifierError::Transcript(
+						ErrorKind::Other,
+						"invalid field element encoding in proof".to_string(),
+					)
+				})
+				.unwrap();
```

Comment:
We should not `unwrap`, but instead propagate error down the stack

Reply:
I pushed a new commit and I tried manual printing for handling it. Is it good way to handle it manually by printing error and returning unknown() value? Value type doesn't allow me to handle it in traditional way.